Add should-fail test for merge with relationship

eileenmcnaughton opened the pull request:
Overview
----------------------------------------
@ejegg  - that getSingle should fail if 2 exist (currently not for me)

Before
----------------------------------------
_What is the old user-interface or technical-contract (as appropriate)?_
_For optimal clarity, include a concrete example such as a screenshot, GIF ([LICEcap](http://www.cockos.com/licecap/), [SilentCast](https://github.com/colinkeenan/silentcast)), or code-snippet._

After
----------------------------------------
_What changed? What is new old user-interface or technical-contract?_
_For optimal clarity, include a concrete example such as a screenshot, GIF ([LICEcap](http://www.cockos.com/licecap/), [SilentCast](https://github.com/colinkeenan/silentcast)), or code-snippet._

Technical Details
----------------------------------------
_If the PR involves technical details/changes/considerations which would not be manifest to a casual developer skimming the above sections, please describe the details here._

Comments
----------------------------------------
_Anything else you would like the reviewer to note_

eileenmcnaughton commented:
@ejegg this is now happily failing :-)

seamuslee001 commented on tests/phpunit/CRM/Dedupe/MergerTest.php:
```diff
@@ -1472,6 +1474,26 @@ public function testMergeWithDeclaredSearchJoin(): void {
+    $this->createCustomGroupWithFieldOfType(['extends' => 'Relationship']);
+    $this->callAPISuccess('Relationship', 'create', [
+      'contact_id_a' => $contact2,
+      'contact_id_b' => CRM_Core_Config::domainID(),
```
@eileenmcnaughton is this a proper contact id value?

eileenmcnaughton replied:
hmm accidentally yes.....

eileenmcnaughton commented:
Correct fail

```
CRM_Dedupe_MergerTest::testMergeWithRelationshipWithCustomFields
Invalid getsingle resultArray
(
    [count] => 2
    [is_error] => 1
    [error_message] => Expected one Relationship but found 2
)

 entity: Relationship . 
 params 
 Array
(
    [contact_id_a] => 3
    [version] => 3

```

eileenmcnaughton commented:
test this please

eileenmcnaughton commented:
I added the fix to this so it should be passing now - but have lost Jenkins record of the last run

eileenmcnaughton commented:
Additional relevant tests are:
```
api_v3_RelationshipTest::testRelationshipCreateDuplicateWithCustomFields2
  
api_v3_RelationshipTest::testRelationshipCreateDuplicateWithCustomFields3
```
  these go through the main `Relationship:create` function which seems to have different expectations to the `Relationship:legacyMultipleCreate` called by the employer update code (called from merge code). I think something weird is going on in `legacyMultipleCreate` so am attempting to divide & conquery via #22385

eileenmcnaughton committed:
Add tests for merging contacts with employer with custom fields

eileenmcnaughton committed:
Fix & simplify duplicate check

The duplicate check has a bug in it but also it's really complicated
give we specifically know the ids & type and not any other values
